OptimizelySDKCore/OptimizelySDKCore/Optimizely.m

```diff
@@ -433,26 +445,54 @@ - (nullable NSString *)variableString:(nonnull NSString *)variableKey
     if ([experimentKeysForLiveVariable count] == 0) {
         NSString *logMessage = [NSString stringWithFormat:OPTLYLoggerMessagesNoExperimentsContainVariable, variableKey];
         [self.logger logMessage:logMessage withLevel:OptimizelyLogLevelWarning];
+        
+        if (callback) {
+            callback(nil);
+        }
+        
         return variable.defaultValue;
     }
     
     for (NSString *experimentKey in experimentKeysForLiveVariable) {
+        
         OPTLYVariation *variation = [self variation:experimentKey
                                              userId:userId
-                                         attributes:attributes
-                                 activateExperiment:activateExperiment];
+                                         attributes:attributes];
+        
         
-        if (variation == nil) {
+        if (variation) {
+            NSString *valueForLiveVariable = [self getValueForLiveVariable:variableId variation:variation];
+            
+            NSString *logMessage = [NSString stringWithFormat:OPTLYLoggerMessagesVariableValue, variableId, valueForLiveVariable];
+            [self.logger logMessage:logMessage withLevel:OptimizelyLogLevelInfo];
+            
+            if (activateExperiment) {
+                [self activate:experimentKey
+                        userId:userId
+                    attributes:attributes
+                      callback:^(NSError *error) {
+                          if (callback) {
+                              callback(error);
+                          }
+                      }];
```

Comment:
Aren't these 5 lines going to be semantically the same as merely coding one line?
```
    callback:callback
```